rust/xaynet-analytics/src/database/common.rs

```diff
@@ -0,0 +1,67 @@
+use isar_core::{
+    error::IsarError,
+    object::{data_type::DataType, object_builder::ObjectBuilder},
+};
+use std::vec::IntoIter;
+
+pub trait IsarAdapter: Sized {
+    fn into_field_properties() -> IntoIter<FieldProperty>;
+
+    fn write_with_object_builder(&self, object_builder: &mut ObjectBuilder);
+}
+
+pub trait Repo<T>
```

Comment:
This should be an abstraction over the database so it should not be coupled with isar. 
For the error we can have our own error that wrap (and implement) a `Box<dyn std::error::Error>` or we can add an associated type that implement that. If we want to be able to return different kind of errors (like `ObjectNotFound`, `IOError` we can only go for the first option.
I don't know how errors are handled in this project so some else can give a better suggestion here.

Reply:
In our `Storage` trait (xaynet_server) we decided to use `anyhow` as error type 
here is the link https://github.com/xaynetwork/xaynet/pull/593#discussion_r517153580

Reply:
ok thanks, done. i went the `anyhow way`. 
hopefully i haven't done a mess of it, please let me know in the next review